webots_ros2_core/README.md

```diff
@@ -0,0 +1,127 @@
+        node_executable='webots_launcher',
+        arguments=arguments = [
+            '--mode=realtime',
+            '--world=' + path_to_webots_world_file
+        ]
+    )
+
+    # Driver node
+    controller = ControllerLauncher(
+        package='webots_ros2_epuck',
+        node_executable='driver',
+        parameters=[{'synchronization': synchronization}]
+    )
+
+    return LaunchDescription([
+        webots,
+        controller,
+        RegisterEventHandler(
+            event_handler=launch.event_handlers.OnProcessExit(
+                target_action=webots,
+                on_exit=[EmitEvent(event=launch.events.Shutdown())],
+            )
+        )
+    ])
+```
+The purpose of the launch file is to start Webots, your driver for Webots and to make sure everything is stopped once Webots closed.
+Then, make sure the driver and the launch file are added to `setup.py`, run `colcon build` and your launch file should be ready to be executed:
+```
+ros2 launch my_webots_driver robot_launch.py
+```
+To extend the ROS interface you should go back to `/my_webots_driver/launch/robot_launch.py` and implement more features.
+For example, in order to add a basic support for [DistanceSensor](https://cyberbotics.com/doc/reference/distancesensor) `MyWebotsDriver` class can be extended as follows:
+```Python
+class MyWebotsDriver(WebotsNode):
+    def __init__(self, args):
+        super().__init__('my_webots_driver', args=args)
+        self.sensor = self.robot.getDistanceSensor('my_distance_sensor')
+        self.sensor.enable(self.timestep)
+        self.sensor_publisher = self.create_publisher(Range, '/my_distance_sensor', 1)
+        self.create_timer(self.timestep * 1e-3, self.publish_sensor_data)
+
+    def publish_sensor_data(self)
+        msg = Range()
+        msg.header.stamp = self.get_clock().now().to_msg()
+        msg.header.frame_id = 'my_distance_sensor'
+        msg.field_of_view = self.sensor.getAperture()
+        msg.min_range = self.sensor.getMinValue()
+        msg.max_range = self.sensor.getMaxValue()
+        msg.range = self.sensor.getValue()
+        msg.radiation_type = Range.INFRARED
+        self.sensor_publisher.publish(msg)
+```
+
+
+### Examples
+This Github repository contains a few good examples that you can use as the starting point:
+- `webots_ros2_example` includes a very simple controller for Thymio (differential driver robot).
+- `webots_ros2_tiago` is another differential drive robot, but here `WebotsDifferentialDriveNode` class from `webots_ros2_core` is utilized to simplify differential drive implementation.
+- `webots_ros2_epuck` is again differential drive robot that creates ROS services and topics for almost all available sensors and actuators.
+This example also contains a [list of instructions](https://github.com/cyberbotics/webots_ros2/blob/master/webots_ros2_epuck/EPUCK_ROS2.md) that explains how the simulation can be used in combination with different ROS2 packages like RViz and Navigation2.
```

Comment:
```suggestion

This example also contains a [list of instructions](https://github.com/cyberbotics/webots_ros2/blob/master/webots_ros2_epuck/EPUCK_ROS2.md) that explains how the simulation can be used in combination with different ROS2 packages like RViz and Navigation2.
```
after a list an empty line is required.

Reply:
Also I never saw `RViz` written with only the R and V as capital, the following seems more commun:
  - rviz
  - RVIZ
  - (Rviz).

Reply:
That capitalization is a huge mess, just on a single page of [the official `rviz` tutorial](http://wiki.ros.org/rviz) they use:
RViz, Rviz and rviz (section `Tutorials`)

In my defence, they use RViz in the logo :)

Reply:
![image](https://user-images.githubusercontent.com/1264964/84359836-7fda2380-abc9-11ea-8015-3de7a0e76ba3.png)
Yes, this is a strong argument for "RViz" indeed...

Reply:
Indeed, then keep it like this :smile: